tests/backup/backup_helper.go

```diff
@@ -4171,3 +4171,38 @@ func CreateRuleForBackupWithMultipleApplications(orgID string, appList []string,
 	}
 	return preRuleName, postRuleName, nil
 }
+
+// DeleteAllAdminBackups deletes all backup from admin
+func DeleteAllAdminBackups() error {
+	var bkp *api.BackupObject
+	backupDriver := Inst().Backup
+	ctx, err := backup.GetAdminCtxFromSecret()
+	if err != nil {
+		return err
+	}
+	bkpEnumerateReq := &api.BackupEnumerateRequest{
+		OrgId: orgID}
+	curBackups, err := backupDriver.EnumerateBackup(ctx, bkpEnumerateReq)
+	if err != nil {
+		return err
+	}
+	for _, bkp = range curBackups.GetBackups() {
+		backupUID, err := backupDriver.GetBackupUID(ctx, bkp.GetName(), orgID)
+		if err != nil {
+			return err
+		}
+		bkpDeleteRequest := &api.BackupDeleteRequest{
+			Name:    bkp.GetName(),
+			OrgId:   orgID,
+			Uid:     backupUID,
+			Cluster: bkp.Cluster,
+		}
+		_, err = backupDriver.DeleteBackup(ctx, bkpDeleteRequest)
+		if err != nil {
+			return err
+		}
+		err = backupDriver.WaitForBackupDeletion(ctx, bkp.GetName(), orgID, backupDeleteTimeout, backupDeleteRetryTime)
+		log.FailOnError(err, "Error waiting for backup deletion %v", bkp.GetName())
+	}
+	return nil
```

Comment:
How about?

```
// DeleteAllUserBackups will delete all user backups
func DeleteAllUserBackups(ctx context.Context, orgId string) error {
	bkpEnumerateReq := &api.BackupEnumerateRequest{
		OrgId: orgId,
	}
	curBackups, err := Inst().Backup.EnumerateBackup(ctx, bkpEnumerateReq)
	if err != nil {
		return err
	}
	errChan := make(chan error, len(curBackups.GetBackups()))
	var wg sync.WaitGroup
	for _, bkp := range curBackups.GetBackups() {
		wg.Add(1)
		go func(bkp *api.BackupObject) {
			defer wg.Done()
			bkpDeleteRequest := &api.BackupDeleteRequest{
				Name:    bkp.GetName(),
				OrgId:   bkp.GetOrgId(),
				Uid:     bkp.GetUid(),
				Cluster: bkp.GetCluster(),
			}
			_, err := Inst().Backup.DeleteBackup(ctx, bkpDeleteRequest)
			if err != nil {
				errChan <- err
				return
			}
			err = Inst().Backup.WaitForBackupDeletion(ctx, bkp.GetName(), bkp.GetOrgId(), backupDeleteTimeout, backupDeleteRetryTime)
			if err != nil {
				errChan <- err
			}
		}(bkp)
	}
	wg.Wait()
	close(errChan)
	var errList []string
	for err := range errChan {
		errList = append(errList, err.Error())
	}
	if len(errList) > 0 {
		return fmt.Errorf(strings.Join(errList, "; "))
	}
	return nil
}

```

Reply:
I have not tested this yet, but I hope it will speed up the process somewhat.

Reply:
done